lib/matplotlib/figure.py

```diff
@@ -2240,24 +2250,32 @@ def __init__(self,
             %(Figure:kwdoc)s
         """
         super().__init__(**kwargs)
+        self._layout_engine = None
 
         if layout is not None:
-            if tight_layout is not None:
+            if (tight_layout is not None or
+                    constrained_layout is not None):
                 _api.warn_external(
-                    "The Figure parameters 'layout' and 'tight_layout' "
-                    "cannot be used together. Please use 'layout' only.")
+                    "The Figure parameters 'layout' and 'tight_layout' or "
+                    "'constrained_layout' cannot be used together. Please use "
+                    "'layout' only.")
+            self.set_layout_engine(layout=layout)
+        elif tight_layout is not None:
             if constrained_layout is not None:
                 _api.warn_external(
-                    "The Figure parameters 'layout' and 'constrained_layout' "
-                    "cannot be used together. Please use 'layout' only.")
-            if layout == 'constrained':
-                tight_layout = False
-                constrained_layout = True
-            elif layout == 'tight':
-                tight_layout = True
-                constrained_layout = False
-            else:
-                _api.check_in_list(['constrained', 'tight'], layout=layout)
+                    "The Figure parameters 'tight_layout' and "
+                    "'constrained_layout' cannot be used together. Please use "
+                    "'layout' parameter")
+            self.set_layout_engine(layout='tight')
+            if isinstance(tight_layout, dict):
+                self.get_layout_engine().set_info(tight_layout)
+        elif constrained_layout is not None:
+            self.set_layout_engine(layout='constrained')
+            if isinstance(constrained_layout, dict):
+                self.get_layout_engine().set_info(constrained_layout)
```

Comment:
What is `set_info`? Seems that it should be `set` (and this is why tests should not have been dropped for the old API, as coverage shows this is not checked.)